pkg/deployments/tasks/scripts/20230116-ta-transition-migrator/input/mainnet.ts

```diff
@@ -0,0 +1,82 @@
+import { flatten } from 'lodash';
+import Task, { TaskMode } from '../../../../src/task';
+import { RoleData } from './types';
+
+const StablePool = new Task('20210624-stable-pool', TaskMode.READ_ONLY, 'mainnet');
+const StablePoolV2 = new Task('20220609-stable-pool-v2', TaskMode.READ_ONLY, 'mainnet');
+const ComposableStablePool = new Task('20220906-composable-stable-pool', TaskMode.READ_ONLY, 'mainnet');
+const ComposableStablePoolV2 = new Task('20221122-composable-stable-pool-v2', TaskMode.READ_ONLY, 'mainnet');
+const WeightedPoolV2 = new Task('20220908-weighted-pool-v2', TaskMode.READ_ONLY, 'mainnet');
+const NoProtocolFeeLbp = new Task('20211202-no-protocol-fee-lbp', TaskMode.READ_ONLY, 'mainnet');
+const ManagedPool = new Task('20221021-managed-pool', TaskMode.READ_ONLY, 'mainnet');
+const AaveRebalancedLinearPool = new Task('20220817-aave-rebalanced-linear-pool', TaskMode.READ_ONLY, 'mainnet');
+const AaveRebalancedLinearPoolV3 = new Task('20221207-aave-rebalanced-linear-pool-v3', TaskMode.READ_ONLY, 'mainnet');
+
+const PoolRecoveryHelper = new Task('20221123-pool-recovery-helper', TaskMode.READ_ONLY, 'mainnet');
+
+const ArbitrumRootGaugeFactoryV2 = new Task('20220823-arbitrum-root-gauge-factory-v2', TaskMode.READ_ONLY, 'mainnet');
+
+const BLABS_OPS_MULTISIG = '0x02f35dA6A02017154367Bc4d47bb6c7D06C7533B';
+const EMERGENCY_SUBDAO_MULTISIG = '0xa29f61256e948f3fb707b4b3b138c5ccb9ef9888';
+const BALLERS_MULTISIG_GAUNTLET = '0xf4a80929163c5179ca042e1b292f5efbbe3d89e6';
+const DAO_MULTISIG = '0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f';
+
+const createRoleData = (grantee: string, target: string, actionIds: string[]): RoleData[] =>
+  actionIds.map((actionId) => ({ role: actionId, grantee: grantee.toLowerCase(), target: target.toLowerCase() }));
+
+// Hard-coded roles taken from https://forum.balancer.fi/t/bip-131-pool-factory-permission-granting/4144.
+export const roles: RoleData[] = flatten([
+  createRoleData(BALLERS_MULTISIG_GAUNTLET, DAO_MULTISIG, [
```

Comment:
This is what the permissions looks like on-chain:
- Grantee (account): ballers multisig
- Target: DAO multisig

I guess the target actually means `EVERYWHERE`, but it's just that the old authorizer doesn't care about the target.

I think the correct thing to do is one of the following:
- Use `EVERYWHERE` as target for all permissions
  - Then, modify the script that fetches the permissions on-chain so that it doesn't fail when comparing this list with the on-chain permissions.
- Use `EVERYWHERE` for permissions that apply to pools (e.g. pause, enable recovery mode, etc) and use factory addresses as targets for permissions (e.g. disable).
  - In this case I'd relax the `target` comparison in the input script, since the old authorizer does not care about it anyways.

In either case, the contract shouldn't need modifications: the target account is not used in the old authorizer's `canPerform`.

Reply:
Yeah, the old authorizer doesn't use `where` so `canPerform` should return the same value no matter what `where` you pass.

I'd opt for the second option, pools should have `EVERYWHERE` as the permission needs to work on all pools. If the permission is for a specific contract then it should be set as the `where` in order to keep the permission as constrained as possible (this isn't really necessary as the action id disambiguator  should handle this as well but it doesn't hurt).

Reply:
on it!

Reply:
targets corrected for each permission in https://github.com/balancer-labs/balancer-v2-monorepo/pull/2151/commits/225d4d76238f588b6583e2b8bfe35dacf1ed7396.